jena-arq/src/main/java/org/apache/jena/riot/system/AsyncParserBuilder.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jena.riot.system;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import org.apache.jena.atlas.iterator.Iter;
+import org.apache.jena.atlas.iterator.IteratorCloseable;
+import org.apache.jena.graph.Triple;
+import org.apache.jena.riot.RDFParserBuilder;
+import org.apache.jena.sparql.core.Quad;
+import org.slf4j.Logger;
+
+public class AsyncParserBuilder {
+    /* These following two attributes only exist for javadoc references */
+    private static final int dftChunkSize = AsyncParser.dftChunkSize;
+    private static final int dftQueueSize = AsyncParser.dftQueueSize;
+
+    protected int chunkSize;
+    protected int queueSize;
+    protected boolean daemonMode;
+    protected List<RDFParserBuilder> sources;
+
+    public AsyncParserBuilder() {
+        this.chunkSize = dftChunkSize;
+        this.queueSize = dftQueueSize;
+        this.daemonMode = true;
+        this.sources = Collections.emptyList();
+    }
+
+    public AsyncParserBuilder(List<RDFParserBuilder> sources) {
+        this();
+        this.sources = sources;
```

Comment:
This can be isolated with `this.sources = List.copyOf(sources);` then changes to the argument have no effect or concurrency issue. WDYT?

Reply:
Fine with me.

Reply:
done